### PR TITLE
[DOCS] Project Export Internals Overview

### PR DESCRIPTION
**Pull request by tkothadev**
## Change summary

Added documentation to the Engine Developer guide giving an overview to the internals of the Project Export mechanism, some of the APIs, and the standard Export Scripts.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Review by Ulrick28 — APPROVED** [review]
Thanks for getting this in!

File: content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+To put this in perspective, back in Q4 2022, an engineer took 1 week to manually package the Newspaper Delivery Game demo project. The same process with the Project Export command takes 2-15 minutes.
+
+The Project Export command also follows the spirit of modularity found in other parts of the O3DE ecosystem, as diverse needs grow overtime in the open source ecosystem.
+
+## Command Setup
+The `export-project` command has the following setup:
+
+![Project Export CLI setup](/images/engine-dev/o3de-cli/project-export/project-export-cli-setup.png)
+
+The central driver of the command is the Export Script. In O3DE, the Export Script is simply any python script that the user wishes to run using the O3DE Python CLI.
```
Comment (AMZN-Gene) [inline content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md:22]: Python proper noun
```suggestion
The central driver of the command is the Export Script. In O3DE, the Export Script is simply any Python script that the user wishes to run using the O3DE Python CLI.
```

File: content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+Note that `export-project` only partially parses arguments. This is because the remaining arguments are fed through to the main Export Script when it runs, which will have its own argument parsing logic.
+
+## Execution Path
+The entry function is called [`_run_export_script`](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/export_project.py#L367), and is structured to start the setup of the export process in the following layers:
+
+##### Layer 1: **`_run_export_script`**  
+Enables logging and sets up the user arguments to be fed into the export system.
+##### Layer 2: **[`_export_script`](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/export_project.py#L307)** 
+Validates the user supplied export script and project, and determine the locations (if the user supplied a relative path). In some cases, this layer will check to see if the user asked for the O3DE standard export scripts.
```
Comment (AMZN-Gene) [inline content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md:34]: What locations? Project directory and export script file location?

File: content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+`export_standalone_run_command` takes the arguments previously specified, validates them, and assuming everything is correct, proceeds to call the `export_standalone_project`.
+
+`export_standalone_project` is the workhorse responsible for preparing the project as a game for PC. These are the main steps in this process: 
+1. Determine Engine type, validate paths and arguments, and determine levels and seedlists to package. [code](https://github.com/o3de/o3de/blob/development/scripts/o3de/ExportScripts/export_source_built_project.py#L84-L128)
```
Comment (AMZN-Gene) [inline content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md:62]:
```suggestion
1. Determine Engine type (source built or SDK installed), validate paths and arguments, and determine levels and seedlists to package. [code](https://github.com/o3de/o3de/blob/development/scripts/o3de/ExportScripts/export_source_built_project.py#L84-L128)
```

**Review by AMZN-Gene — APPROVED** [review]

File: content/docs/engine-dev/tools/o3de-cli/project-export/_index.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+---
+linkTitle: Project Export
+title: Project Export
+description: High Level Developer Overview for the Project Export command.
+---
+
+***The Project Export*** command enables shipping of O3DE projects into distributable games. It is designed to fulfill arbitrary needs in the export process, and comes with standard export scripts for the majority of users.
+
+{{< note >}}
+You can find a presentation of the Project Export feature on the official O3DE Youtube channel: https://www.youtube.com/watch?v=fV1gmxiRHdY
```
Comment (forkercat) [inline content/docs/engine-dev/tools/o3de-cli/project-export/_index.md:10]: -> YouTube

File: content/docs/engine-dev/tools/_index.md [hunk 5]
```diff
@@ -17,3 +17,4 @@ This is a stub page intended to serve as an example of guide structure. Informat
 |-|-|
 | [Project Manager](./project-manager) | Learn about the internals of **Project Manager**, including an overview of the major classes and data structures that work together to create the UI and implement the project management and Gem library features of this tool. |
 | [Test Impact Analysis Framework](./tiaf) | Learn about the internals of the Test Impact Analysis Framework, including an overview of the major classes and data structures that work together to create the C++ and Python test support of this tool. |
+| [O3DE CLI](./o3de-cli) | Learn about the internals of the **O3DE CLI**, . |
```
Comment (forkercat) [inline content/docs/engine-dev/tools/_index.md:20]: extra comma after "CLI"

File: content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+The way this works is that O3DE CLI (`o3de.bat` on Windows or `o3de.sh` on Linux) [invokes](https://github.com/o3de/o3de/blob/development/scripts/o3de.py#L107-L108) the `export-project` [sub-command argument](https://github.com/o3de/o3de/blob/development/scripts/o3de.py#L79) via a specified [entry-hook](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/export_project.py#L484). This system heavily relies on the `argparse` standard library to configure the expected function for the parsed command in a modular fashion.
+
+Note that `export-project` only partially parses arguments. This is because the remaining arguments are fed through to the main Export Script when it runs, which will have its own argument parsing logic.
+
+## Execution Path
+The entry function is called [`_run_export_script`](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/export_project.py#L367), and is structured to start the setup of the export process in the following layers:
+
+##### Layer 1: **`_run_export_script`**
```
Comment (forkercat) [inline content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md:31]: For each layer title, maybe consider using "###".

File: content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+The way this works is that O3DE CLI (`o3de.bat` on Windows or `o3de.sh` on Linux) [invokes](https://github.com/o3de/o3de/blob/development/scripts/o3de.py#L107-L108) the `export-project` [sub-command argument](https://github.com/o3de/o3de/blob/development/scripts/o3de.py#L79) via a specified [entry-hook](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/export_project.py#L484). This system heavily relies on the `argparse` standard library to configure the expected function for the parsed command in a modular fashion.
+
+Note that `export-project` only partially parses arguments. This is because the remaining arguments are fed through to the main Export Script when it runs, which will have its own argument parsing logic.
+
+## Execution Path
+The entry function is called [`_run_export_script`](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/export_project.py#L367), and is structured to start the setup of the export process in the following layers:
+
+##### Layer 1: **`_run_export_script`**  
+Enables logging and sets up the user arguments to be fed into the export system.
```
Comment (forkercat) [inline content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md:32]: "... sets up to feed the user arguments into the export system."

File: content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+This runs an instance of the [`CLICommand`](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/utils.py#L68) object with the supplied arguments and working directory. This effectively calls `subprocess.Popen`, and wraps all CLI related execution into a [`busy loop`](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/utils.py#L110-L116) until the process is done. It also wraps all stdout and stderr logs and feeds it forward to the Export Script so that the user sees all logs as the command proceeds.
+
+Wrapping the subprocess in this manner allows us to [clean up](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/utils.py#L118-L124) and safely terminate all processes invoked.
+
+
+##### **[`build_assets`](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/export_project.py#L583)** 
+This first calculates the path of the `AssetProcessorBatch` executable, based on whether O3DE is a source built or SDK installed engine. Based on that `process_command` is used to run the batch tool with arguments for the project, and selected platform if specified.
```
Comment (forkercat) [inline content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md:80]: Do you mean "for the project, and with selected platform"?

File: content/docs/engine-dev/tools/o3de-cli/project-export/_index.md (hunk 4 above)
Comment (tkothadev) [reply content/docs/engine-dev/tools/o3de-cli/project-export/_index.md:10]: do you mean change "Youtube channel:" to "Youtube:"?

File: content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md (hunk 6 above)
Comment (tkothadev) [reply content/docs/engine-dev/tools/o3de-cli/project-export/high-level-overview.md:31]: That was what I originally wanted to do, but for each section heading that had an embedded link, markdown caused it to become transparent unless you moused over it. This was the only header style for the layer title that did not cause titles with links to disappear.

**Pull request merged**